include/ada/scheme-inl.h

```diff
@@ -16,46 +16,50 @@ namespace ada::scheme {
 namespace details {
 // for use with is_special and get_special_port
 // Spaces, if present, are removed from URL.
-constexpr std::string_view is_special_list[] = {"http", " ",   "https", "ws",
-                                                "ftp",  "wss", "file",  " "};
+static constexpr std::string_view is_special_list[] = {
+    "http", " ", "https", "ws", "ftp", "wss", "file", " "};
+
 // for use with get_special_port
-constexpr uint16_t special_ports[] = {80, 0, 443, 80, 21, 443, 0, 0};
+static constexpr uint16_t special_ports[] = {80, 0, 443, 80, 21, 443, 0, 0};
 }  // namespace details
 
 ada_really_inline constexpr bool is_special(std::string_view scheme) {
   if (scheme.empty()) {
     return false;
   }
-  int hash_value = (2 * scheme.size() + (unsigned)(scheme[0])) & 7;
+  auto const hash_value = static_cast<int>(
+      (2 * static_cast<int>(scheme.size()) + static_cast<unsigned>(scheme[0])) &
+      7U);
   const std::string_view target = details::is_special_list[hash_value];
```

Comment:
You are replacing 
```C++
 int hash_value = (2 * scheme.size() + (unsigned)(scheme[0])) & 7;
```
with 
```
auto const hash_value = static_cast<int>(
       (2 * static_cast<int>(scheme.size()) + static_cast<unsigned>(scheme[0])) &
       7U);
```

Please explain when the two expressions would give different results.

Reply:
I'm simply avoiding c-style casts, and intentionally express the casts in order to **avoid implicit casts**.

This should hail the same result as the previous code, if it doesn't it's a mistake from me.